src/main/java/com/as/plugins/customfields/jira/customfields/WSJFField.java

```diff
@@ -0,0 +1,96 @@
+import com.atlassian.jira.issue.customfields.SortableCustomField;
+import com.atlassian.jira.issue.customfields.impl.CalculatedCFType;
+import com.atlassian.jira.issue.customfields.impl.FieldValidationException;
+import com.atlassian.jira.issue.CustomFieldManager;
+import com.atlassian.jira.issue.fields.CustomField;
+import com.atlassian.jira.issue.Issue;
+import com.atlassian.jira.issue.MutableIssue;
+import com.atlassian.jira.issue.fields.CustomField;
+import com.atlassian.jira.issue.fields.config.FieldConfig;
+import com.atlassian.jira.issue.fields.config.FieldConfigItemType;
+import com.atlassian.jira.issue.fields.layout.field.FieldLayoutItem;
+import com.atlassian.plugin.spring.scanner.annotation.component.Scanned;
+import java.util.List;
+import java.util.Map;
+
+@Scanned
+public class WSJFField extends CalculatedCFType implements SortableCustomField{
+    // Set the custom field ids to the strings below
+    private String storyPointsId = "customfield_10008";
+    private String timeCriticalityId = "customfield_14515";
+    private String riskOpportunityValueId = "customfield_14512";
+    private String businessValuePreId = "customfield_14503";
+
+    @Override
+    public Map<String, Object> getVelocityParameters(final Issue issue,
+                                                     final CustomField field,
+                                                     final FieldLayoutItem fieldLayoutItem) {
+        final Map<String, Object> map = super.getVelocityParameters(issue, field, fieldLayoutItem);
+
+        // This method is also called to get the default value, in
+        // which case issue is null so we can't use it to add currencyLocale
+        if (issue == null) {
+            return map;
+        }
+
+         FieldConfig fieldConfig = field.getRelevantConfig(issue);
+         //add what you need to the map here
+
+        return map;
+    }
+
+    @Override
+    public Object getValueFromIssue(com.atlassian.jira.issue.fields.CustomField field1,com.atlassian.jira.issue.Issue issue1) {
+        //MutableIssue issueM = (MutableIssue)issue1;
+        CustomFieldManager customFieldManager = ComponentAccessor.getCustomFieldManager();
+        CustomField cf1 = customFieldManager.getCustomFieldObject(timeCriticalityId);
+        CustomField cf2 = customFieldManager.getCustomFieldObject(riskOpportunityValueId);
+        CustomField cf3 = customFieldManager.getCustomFieldObject(businessValuePreId);
+        CustomField cf4 = customFieldManager.getCustomFieldObject(storyPointsId);
+        Object o1 = issue1.getCustomFieldValue(cf1);
+        Object o2 = issue1.getCustomFieldValue(cf2);
+        Object o3 = issue1.getCustomFieldValue(cf3);
+        Object o4 = issue1.getCustomFieldValue(cf4);
+
+        if (o1 == null
+            || o2 == null
+            || o3 == null
+            || o4 == null) {
+            return -1.00;
+        }
+
+        Double value1 = Double.parseDouble(o1.toString());
```

Comment:
I know that in the instructions, you indicate that the fields must be either numbers or a select, and presumably values in the select must also be numbers. I'm curious what happens, though, should someone **not** follow those instructions and instead have them as string fields, and then, should a user enter in a string like "foobar"... this code will error. My question is: in that scenario, when an error is thrown, what happens on screen? Presumably the effect would be limited to just that user's interaction with the screen and not more widespread than that, but I'm curious if you've tested that condition

Reply:
Yeah there are three things that happen in this scenario, when you edit the record in the full edit view and try to save it with a non numerical value in one of those fields, it will not let the record save and there is no error saying why. If you in line edit and try to save it with a non numerical value in one of those fields a little popup in the corner appears saying there was a java.lang.NumberFormatException, but the record does save. Then if you reload the page that was the bad value an error message popups up saying "There were errors loading the issue" but the whole page loads leaving off the WSJF field.